libmpeg2: fix distfiles.

JkktBkkt opened the pull request:
~~Internals are identical to those of the original source (found on archive.org), according to diff -r~~
EDIT: switched to debian-hosted distfiles, hash matches previous version. Added patch of test from [gentoo](https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/libmpeg2/files/libmpeg2-0.5.1-global-symbol-test.patch)

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x86-64

slymattz commented:
I think you'll have to do the following;
```
case "$XBPS_TARGET_MACHINE" in
	ppc64*) ;;
	ppc*) CFLAGS="-mno-altivec" ;;
	i686*) make_check=no ;;
esac
```
I ran into do_check() problems for i686 locally, so this doesn't seem like a CI issue. Oh, and don't forget to squash your commits into one.

slymattz commented:
> ~Internals are identical to those of the original source (found on archive.org), according to diff -r~ EDIT: switched to debian-hosted distfiles, hash matches previous version. Added patch of test from [gentoo](https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/libmpeg2/files/libmpeg2-0.5.1-global-symbol-test.patch)

It seems you forgot to add the patch. Anyway, remember to squash your commits into one.

JkktBkkt commented:
> > ~Internals are identical to those of the original source (found on archive.org), according to diff -r~ EDIT: switched to debian-hosted distfiles, hash matches previous version. Added patch of test from [gentoo](https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/libmpeg2/files/libmpeg2-0.5.1-global-symbol-test.patch)
> 
> It seems you forgot to add the patch. Anyway, remember to squash your commits into one.

You're exactly right, thanks. 
I typically commit locally with --amend for simpler things like small template fixes and tab-completed just the template instead of passing the dir to it, my bad
